optuna/study/study.py

```diff
@@ -462,6 +462,18 @@ def ask(
             A :class:`~optuna.trial.Trial`.
         """
 
+        if self._optimize_lock.acquire(False):
+            self._optimize_lock.release()
```

Comment:
[Question] What is the difference between this and checking `self._optimize_lock.locked`?
https://docs.python.org/3/library/threading.html#threading.Lock.locked

Reply:
Thank you for your suggestion. That makes sense. We don't need to acquire lock here because we just wanted to know the status of the lock. I fixed it in https://github.com/optuna/optuna/pull/3273/commits/d70c2bb59d3650a1dc6b33eb325ff6269955c373.